ios/RNIapIos.m

```diff
@@ -95,12 +92,14 @@ -(void)rejectPromisesForKey:(NSString*)key code:(NSString*)code message:(NSStrin
     }
 }
 
-////////////////////////////////////////////////////     _//////////_  // IAPPromotionObserverDelegate
-- (BOOL)shouldAddStorePayment:(SKPayment *)payment forProduct:(SKProduct *)product {
-    if (hasListeners) {
-        [self sendEventWithName:@"iap-promoted-product" body:product.productIdentifier];
-    }
-    return NO;
+- (BOOL)paymentQueue:(SKPaymentQueue *)queue shouldAddStorePayment:(SKPayment *)payment forProduct:(SKProduct *)product {
+  promotedProduct = product;
+  promotedPayment = payment;
```

Comment:
This code is merged from the following code.

https://github.com/dooboolab/react-native-iap/blob/3f103f20c674c091fa69b715102bfc4caac37462/ios/IAPPromotionObserver.m#L50-L52